logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/local/LocalEventManager.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.sync.local
+
+import com.wire.kalium.common.functional.Either
+import com.wire.kalium.common.logger.kaliumLogger
+import com.wire.kalium.logic.data.event.EventEnvelope
+import com.wire.kalium.logic.sync.incremental.EventProcessor
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * LocalEventManager listens for local events from LocalEventRepository and processes them using EventProcessor.
+ */
+internal interface LocalEventManager {
+    /**
+     * Starts processing local events.
+     */
+    fun startProcessing()
+}
+
+internal class LocalEventManagerImpl(
+    private val localEventRepository: LocalEventRepository,
+    private val eventProcessor: EventProcessor
+) : LocalEventManager, CoroutineScope {
+
+    override val coroutineContext: CoroutineContext = SupervisorJob() + Dispatchers.IO
```

Comment:
here we can use the user scope CoroutineScope since it will also be canceled on logout